pytorch_lightning/trainer/train_loader_patch.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any, Union
+from collections.abc import Iterable, Mapping, Sequence
+
+from pytorch_lightning.utilities.data import get_len
+from pytorch_lightning.utilities.apply_func import apply_to_collection
+
+
+class MultiIterator(object):

```

Comment:
Where is this class used outside the tests ?

Reply:
good catch. This was an early version of the `CombinedIterator` above. Will remove this!